packages/trader/src/AppV2/Utils/symbol-categories-utils.ts

```diff
@@ -0,0 +1,102 @@
+import { ActiveSymbols } from '@deriv/api-types';
+import { localize } from '@deriv/translations';
+
+type SubmarketGroup = {
+    submarket_display_name: string;
+    items: ActiveSymbols;
+};
+
+type SubgroupGroup = {
+    subgroup_display_name: string;
+    submarkets: Record<string, SubmarketGroup>;
+};
+
+export type MarketGroup = {
+    market: string;
+    market_display_name: string;
+    subgroups: Record<string, SubgroupGroup>;
+};
+
+export const categorizeSymbols = (symbols: ActiveSymbols): Record<string, MarketGroup> => {
+    if (symbols.length === 0) {
+        return {};
+    }
+    // Categorize ActiveSymbols array into object categorized by markets
+    let categorizedSymbols = symbols.reduce((acc: Record<string, MarketGroup>, symbol: ActiveSymbols[0]) => {
+        const { market, market_display_name, subgroup, subgroup_display_name, submarket, submarket_display_name } =
+            symbol;
+
+        if (!acc[market]) {
+            acc[market] = {
+                market,
+                market_display_name,
+                subgroups: {},
+            };
+        }
```

Comment:
```suggestion
       acc[market] ??= { market, market_display_name, subgroups: {} };
```
Can we simplify this and the below conditions with a logical nullish assignment?